### PR TITLE
GOV.UK Framework | Staff Record | Main Job Start Date

### PR DESCRIPTION
**Pull request by ChazUK**
This PR touches a number of files and areas of the Staff Record Flow.

* Updated Date Validators to be more consistent and use built in `moment` functions
* Updated Date Picker Component so that it can be imported and used multiple times on a page
* Created Base Question Component that pulls out all the duplicate code from the Staff Records Questions into one class
* Removing unused SCSS that has been trailing in this branch
* Added a utils file for reusable SCSS
* Allow users to be redirected to different pages as a `return` function from Question Pages. This can either be the Staff Record page or the Check Answers Summary
* Submit Buttons Shared Component now emits events to the parent component to let them handle the logic

### CODE REVIEW

File: src/app/core/services/worker.service.ts [hunk 1]
```diff
@@ -77,8 +77,8 @@ export class WorkerService {
     this._worker$.next(worker);
   }
 
-  setReturnToSummary(val) {
-    this.returnToSummary$.next(val);
+  setReturnTo(val: string[]) {
```
Comment (omarCreativeDev) [inline src/app/core/services/worker.service.ts]: is it possible to type the function as well as its return type?

Comment (ChazUK) [reply src/app/core/services/worker.service.ts]: Sure `void` ok?

File: src/app/features/workers/main-job-start-date/main-job-start-date.component.html [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
-  </form>
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds-from-desktop">
+    <h1 class="govuk-heading-xl">What date did they start their main job role?</h1>
+  </div>
 </div>
+<form (ngSubmit)="onSubmit()" [formGroup]="form">
+  <div class="govuk-form-group" [class.govuk-form-group--error]="submitted && (form.invalid || serverError)">
+    <fieldset class="govuk-fieldset" aria-describedby="mainJobStartDate-hint" role="group">
+      <!-- <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
```
Comment (omarCreativeDev) [inline src/app/features/workers/main-job-start-date/main-job-start-date.component.html]: how comes this is commented out?

File: src/app/core/services/worker.service.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/app/core/services/worker.service.ts]: sweet

File: src/app/features/workers/main-job-start-date/main-job-start-date.component.html (hunk 2 above)
Comment (ChazUK) [reply src/app/features/workers/main-job-start-date/main-job-start-date.component.html]: Good spot, wanted to discuss this with Rich

**Comment by ChazUK** [comment]
[Staff Record - Main Job Start Date](https://trello.com/c/R9rjT0Td/508-staff-record-main-job-start-date)

**Pull request merged**